lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -112,4 +120,107 @@ module.exports = {
+          const functionObj = this.serverless.service.getFunction(functionName);
+
+          if (!functionObj.events) {
+            return BbPromise.resolve();
+          }
+
+          const promises = functionObj.events.map((event) => {
+            if (!event.cloudwatchLog) {
+              return BbPromise.resolve();
+            }
+
+            let logGroupName;
+
+            /*
+              it isn't necessary to run sanity checks here as they already happened during the
+              compile step
+            */
+            if (typeof event.cloudwatchLog === 'object') {
+              logGroupName = event.cloudwatchLog.logGroup.replace(/\r?\n/g, '');
+            } else {
+              logGroupName = event.cloudwatchLog.replace(/\r?\n/g, '');
+            }
+
+            /*
+              return a new promise that will check the resource limit exceeded and will fix it if
+              the option is enabled
+            */
+            return this.fixLogGroupSubscriptionFilters({
+              cloudWatchLogsSdk,
+              accountId,
+              logGroupName,
+              functionName,
+              region,
+              serviceName,
+              stage,
+            });
+          });
+
+          return Promise.all(promises);
+        })));
+  },
+
+  fixLogGroupSubscriptionFilters(params) {
+    const cloudWatchLogsSdk = params.cloudWatchLogsSdk;
+    const accountId = params.accountId;
+    const logGroupName = params.logGroupName;
+    const functionName = params.functionName;
+    const region = params.region;
+    const serviceName = params.serviceName;
+    const stage = params.stage;
+
+    return cloudWatchLogsSdk.describeSubscriptionFilters({ logGroupName }).promise()
+      .then((response) => {
+        const subscriptionFilter = response.subscriptionFilters[0];
+
+        // log group doesn't have any subscription filters currently
+        if (!subscriptionFilter) {
+          return false;
+        }
+
+        const oldDestinationArn = subscriptionFilter.destinationArn;
+        const filterName = subscriptionFilter.filterName;
+        const newDestinationArn =
+          `arn:aws:lambda:${region}:${accountId}:function:${serviceName}-${stage}-${functionName}`;
+
+        // everything is fine, just return
+        if (oldDestinationArn === newDestinationArn) {
+          return false;
+        }
+
+        /*
+          If the destinations functions' ARNs doesn't match, we need to delete the current
+          subscription filter to prevent the resource limit exceeded error to happen
+        */
+        return cloudWatchLogsSdk
+          .deleteSubscriptionFilter({ logGroupName, filterName }).promise();
+      })
+      /*
+        it will throw when trying to get subscription filters of a log group that was just added
+        to the serverless.yml (therefore not created in AWS yet), we can safely ignore this error
+      */
+      .catch(() => undefined);
```

Comment:
Oh, nevermind, this addresses my 'one last thought', right?

Reply:
Yes, that's right.